src/validators/toThrow.ts

```diff
@@ -1,28 +1,52 @@
 import { assert } from 'ts-essentials'
 
 import { Control } from './common'
+import { smartEq } from './toEqual'
 
-export function toThrow(control: Control<() => any>, expectedMsg?: string) {
+export function toThrow(control: Control<() => any>, expected?: any) {
   assert(control.actual instanceof Function, 'Actual has to be a function to check if threw')
 
-  let error: any | undefined
+  let actualThrowValue: any | undefined
+  let threwAnything = false
   try {
     control.actual()
   } catch (e) {
-    error = e
+    threwAnything = true
+    actualThrowValue = e
   }
 
-  if (expectedMsg !== undefined) {
-    control.assert({
-      success: error?.message === expectedMsg,
-      reason: `Expected to throw "${expectedMsg}" but threw "${error?.message}"`,
-      negatedReason: `Expected not to throw "${expectedMsg}" but did`,
+  // we need special handling for this case otherwise we end up with really dummy error message
+  if (!threwAnything) {
+    return control.assert({
+      success: false,
+      reason: `Expected to throw but didn't`,
+      negatedReason: '-',
     })
+  }
+
+  const reason = `Expected to throw "${expected}" but threw "${actualThrowValue}"`
+  const negatedReason = `Expected not to throw "${expected}" but threw "${actualThrowValue}"`
+
+  if (!smartEq(actualThrowValue, expected)) {
+    if (arguments.length === 1 && !control.isNegated) {
+      control.autofix('toThrow', control.actual)
+      control.assert({
```

Comment:
I just realised that assert after autofix is pointless (just the same in other matchers)